lib/plugins/aws/provider/awsProvider.test.js

```diff
@@ -51,6 +53,73 @@ describe('AwsProvider', () => {
       delete process.env.proxy;
     });
 
+    it('should set AWS ca single', () => {
+      process.env.ca = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+
+      // clear env
+      delete process.env.ca;
+    });
+
+    it('should set AWS ca multiple', () => {
+      const certContents = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      process.env.ca = `${certContents},${certContents}`;
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+
+      // clear env
+      delete process.env.ca;
+    });
+
+    it('should set AWS cafile single', () => {
+      const tmpdir = os.tmpdir();
+      const certContents = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      fs.writeFileSync(`${tmpdir}/ca.txt`, certContents);
+      process.env.cafile = `${tmpdir}/ca.txt`;
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+
+      // clear env
+      fs.unlink(`${tmpdir}/ca.txt`);
+      delete process.env.cafile;
+    });
+
+    it('should set AWS cafile multiple', () => {
+      const tmpdir = os.tmpdir();
+      const certContents = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      fs.writeFileSync(`${tmpdir}/ca1.txt`, certContents);
+      fs.writeFileSync(`${tmpdir}/ca2.txt`, certContents);
+      process.env.cafile = `${tmpdir}/ca1.txt,${tmpdir}/ca2.txt`;
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+
+      // clear env
+      fs.unlink(`${tmpdir}/ca1.txt`);
```

Comment:
This should be done in an `afterEach` block.

Reply:
I'm new to Mocha so I might be wrong here but in my reading of the docs, this won't work.

Given the following:
- A hook should be defined in the describe block or at the root level. 
- fs.unlink() throws an exception if the file is not found
- not all tests in the constructor's describe() block require these files so the files will not exist

I don't think this will work. Thoughts?